### DISCUSSION TITLE
Anything stopping this from running on iOS?

### DISCUSSION THREAD

**Discussion by warvstar** [wasmerio/wasmer#1408] [Ideas]
Can wasmer compile and run on iOS using AOT modules(module caches)?

~~Is there anything preventing wasmer from been compiled and run on the web?~~

Edit: sorry my second question was for another runtime, I remember already seeing wasmer in the browser.

Thanks

**Comment by igrep** [comment]
I believe this is just because no one hasn't addressed to run wasmer on iOS.
And even we don't know whether there are any restrictions preventing wasmer from working on iOS.
I'm not a core member of wasmer, but I think they would welcome a patch.

**Comment by igrep** [comment]
Side note: wasmer on Web doesn't actually share any codes with the native wasmer.
Because current wasmer has only JIT compilers: there are no way to run on browsers!

**Comment by vshymanskyy** [comment]
@warvstar did you try [`Wasm3`](https://github.com/wasm3/wasm3) on `iOS` / `Web`? We do support this out of the box, and your feedback would be welcome.

**Comment by warvstar** [comment]
> @warvstar did you try [`Wasm3`](https://github.com/wasm3/wasm3) on `iOS` / `Web`? We do support this out of the box, and your feedback would be welcome.

I have and it works great, now I'm just looking at alternative ways to also compile AOT to get that last bit of performance for my release builds.

I'm considering using wabts wasm2c to transpile then just use clang to compile natively, but I have no idea how that even works with the module imports / exports yet, will look into that today.

To me it looks like what I need doesn't exist yet, so now I just need to find the closest / easiest project to fork to add such capability, as I don't want to spend more time on it than I need to.
